tidy3d/plugins/adjoint/utils/penalty.py

```diff
@@ -70,6 +79,17 @@ def evaluate(self, points: ArrayFloat2D) -> float:
         desired minimum value. If ``wrap``, it is assumed that the points wrap around to form a
         closed geometry instead of an isolated line segment."""
 
+        if not is_jax_object(points):
+            log.warning(
+                "The points passed to 'RadiusPenalty.evaluate()' are not a 'jax' array. "
+                "If you are passing the 'JaxPolySlab.vertices' field directly, note that the "
+                "derivative information for this field is no longer traced by as of 'tidy3d 2.7'. "
+                "The derivative information is contained in 'JaxPolySlab.vertices_jax'. "
+                " Therefore, we recommend changing your code to either pass that field or pass "
```

Comment:
Double space at the end of previous line and at the beginning of this one.

Reply:
fixed